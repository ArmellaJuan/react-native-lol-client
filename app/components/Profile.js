import React, { Component, PropTypes } from 'react';
import { StyleSheet, ScrollView, ActivityIndicator, View, Button, TextInput } from 'react-native';

import SummonerDetails from './SummonerDetails';

export default class Profile extends Component {

  constructor(props) {
    super(props);
    this.onMatchHistory = this.onMatchHistory.bind(this);
  }


  onMatchHistory() {
    this.props.navigator.push({
      title:this.props.name + "'s recent games",
      id: 'matchHistory'
    }); 
  }

  summonerDetails() {
    
    if(this.props.loading){
      return (<ActivityIndicator
            style={[styles.loading]}
          />);
    } 

    if(!this.props.found){
      return (<View />);
    }

    return( 
      <SummonerDetails
       onMatchHistory={this.onMatchHistory}
       summoner={this.props.summoner}
       />
    );
  }


  render() {
    return (
          <View style={styles.profile}>
            <TextInput
                style={{height: 40, backgroundColor: 'white', elevation   : 3}}
                value={this.props.name}
                onChangeText={ this.props.onChangeSummonerName }
                placeholder="Summoner Name"
              />
            <ScrollView>
              <View style={styles.container}>             
                  {this.summonerDetails()}
              </View>
            </ScrollView>
            <Button
                    onPress={ () => this.props.onFetchSummoner(this.props.name) }
                    title="Check Summoner"
                    color="dodgerblue"
                  />
          </View>
    );
  }
}


Profile.propTypes = {
  navigator: PropTypes.object.isRequired,
  onChangeSearchSummonerName: PropTypes.func.isRequired,
  onFetchSummoner: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  found: PropTypes.bool.isRequired,
  summoner: PropTypes.object,
  name: PropTypes.string.isRequired
};


const styles = StyleSheet.create({
  rowInfoLabel: {
    flex: 1
  },
  loading: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 8,
    height: 80
  },
  centering: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 8,
  },
  profile: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: 'aliceblue'
  },
  container: {
    flex: 1,
    padding: 20,
    flexDirection: 'column'
  }
});

