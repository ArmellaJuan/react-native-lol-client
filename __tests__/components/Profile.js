import 'react-native';
import React from 'react';
import Profile from '../../app/components/Profile';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('../../app/components/SummonerDetails', () => 'SummonerDetails');


test('renders correctly', () => {

  let summoner = {
    id: 55899,
    name: "Test"
  };

  const tree = renderer.create(
    <Profile summoner = { summoner } found = { true } searched = {false} loading = { false } name = 'Test' onFetchSummoner = { () => {} }
      onChangeSearchSummonerName = { () => {} }  onRecentGames= {() => {} }/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});