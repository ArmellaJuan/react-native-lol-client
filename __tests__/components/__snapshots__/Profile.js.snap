exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "aliceblue",
      "flex": 1,
      "flexDirection": "column",
    }
  }>
  <TextInput
    onChangeText={undefined}
    placeholder="Summoner Name"
    style={
      Object {
        "backgroundColor": "white",
        "elevation": 3,
        "height": 40,
      }
    }
    value="Test" />
  <ScrollView>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "padding": 20,
        }
      }>
      <SummonerDetails
        onMatchHistory={[Function]}
        summoner={
          Object {
            "id": 55899,
            "name": "Test",
          }
        } />
    </View>
  </ScrollView>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Array [
          Object {},
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "dodgerblue",
            },
          ]
        }>
        Check Summoner
      </Text>
    </View>
  </View>
</View>
`;
